### PR TITLE
Add new react typescript example

### PR DESCRIPTION
**Pull request by BarryThePenguin**

### CODE REVIEW

File: react-typescript/src/mirage/index.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        get password() {
+          return faker.internet.password();
+        },
+        get email() {
+          return faker.internet.email(this.firstName, this.lastName);
+        },
+        get avatar() {
+          return faker.internet.avatar();
+        },
+      }),
```
Comment (BarryThePenguin) [inline react-typescript/src/mirage/index.ts:43]: This isn't entirely correct..

Might need some adjustment to the miragejs types to support something here 👀 

If I remove the getters, the `Person` type doesn't align with how `Factory` works, specifically when accessing `this.firstName`, and `this.lastName`, when the person is first being created

File: react-typescript/src/mirage/index.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    },
+
+    models: {
+      person: Model.extend<Partial<Person>>({}),
```
Comment (BarryThePenguin) [inline react-typescript/src/mirage/index.ts:47]: I used `Partial` here because `Model` expects all the properties of `Person` to be defined

**Comment by samselikoff** [comment]
Awesome, thanks again for this!

Should I merge just so we have something, or do you want me to try to find some folks that might be able to help address your concern above?

**Comment by samselikoff** [comment]
(I'm just not in a place with TS yet to be able to judge this on my own)

**Comment by haluvibe** [comment]
I would recommend merging this now, we can improve this later in a seperate PR. At least this gets us started.

**Comment by samselikoff** [comment]
Awesome, thanks again for kicking this off!

**Pull request merged**